PoseEstimation-TFLiteSwift/OpenPose/OpenPosePoseEstimator.swift

```diff
@@ -193,29 +269,161 @@ private extension PoseEstimationOutput {
     }
     
     func parseMultiHuman(from outputs: [TFLiteFlatArray<Float32>], on partIndex: Int?, with threshold: Float?) -> [Human] {
-        guard let partIndex = partIndex else { return [] }
-        
-        let output = outputs[0] // openpose_ildoonet.tflite only use the first output
+        // openpose_ildoonet.tflite only use the first output
+        let output = outputs[0]
         
-        // 1. nms
+        if let partIndex = partIndex {
+            return parseSinglePartOnMultiHuman(from: output, on: partIndex, with: threshold)
+        } else {
+            return parseAllPartOnMultiHuman(from: output, with: threshold)
+        }
+    }
+    
+    func parseSinglePartOnMultiHuman(from output: TFLiteFlatArray<Float32>, on partIndex: Int, with threshold: Float?) -> [Human] {
+        // process NMS
         let keypointIndexes = output.keypoints(partIndex: partIndex, filterSize: 3, threshold: threshold)
+        
+        // convert col,row to Keypoint
         let kps: [Keypoint] = keypointIndexes.map { keypointInfo in
-            let x = (CGFloat(keypointInfo.col) + 0.5) / CGFloat(OpenPosePoseEstimator.Output.ConfidenceMap.width)
-            let y = (CGFloat(keypointInfo.row) + 0.5) / CGFloat(OpenPosePoseEstimator.Output.ConfidenceMap.height)
-            let score = Float(keypointInfo.val)
-            return Keypoint(position: CGPoint(x: x, y: y), score: score)
+            return Keypoint(column: keypointInfo.col,
+                            row: keypointInfo.row,
+                            width: OpenPosePoseEstimator.Output.ConfidenceMap.width,
+                            height: OpenPosePoseEstimator.Output.ConfidenceMap.height,
+                            value: keypointInfo.val)
         }
         
+        // Make [Human]
         return kps.map { keypoint in
-            let keypoints: [Keypoint] = OpenPosePoseEstimator.Output.BodyPart.allCases.enumerated().map { offset, _ in
-                return (offset == partIndex) ? keypoint : Keypoint(position: .zero, score: -100)
+            let keypoints: [Keypoint?] = OpenPosePoseEstimator.Output.BodyPart.allCases.enumerated().map { offset, _ in
+                return (offset == partIndex) ? keypoint : nil
             }
             return Human(keypoints: keypoints, lines: [])
         }
+    }
+    
+    func parseAllPartOnMultiHuman(from output: TFLiteFlatArray<Float32>, with threshold: Float?) -> [Human] {
```

Comment:
I recommend to divide 'parseAllPartOnMultiHuman' to tasks, that have same abstract level.

example)
```swift
func parseSinglePartOnMultiHuman() {
   ...
   makePair()
   makeEdgesForEachPair()
   merge()
   makeHumans()
}
```

Reply:
This suggestion is really important, but I'll refactoring on next PR soon.